### ISSUE TITLE
TypeError: Cannot read property 'login' of null

### ISSUE THREAD

**Issue by maoo** [finos/cla-bot#74]
The error occurs [on line 134](https://github.com/ColinEberhardt/cla-bot/blob/develop/cla-bot/index.js#L134), which was introduced by issue #70 .

Basically, there are some PRs that come with no `author` or `committer` values, therefore it fails with the error mentioned on the title; an example of such PR is https://github.com/symphonyoss/SymphonyElectron/pull/195 (JSON reported below, emails hidden)

Any idea why this happens?

```
{
    "time": "2017-09-28T13:30:44.627Z",
    "correlationKey": "39f34540-a451-11e7-ba7e-71a0befdf45c",
    "level": "DEBUG",
    "message": "API Response https://api.github.com/repos/symphonyoss/SymphonyElectron/pulls/195/commits",
    "detail": [
        {
            "sha": "1ff558af8121487cdbc0302b4b8efb8e930b33af",
            "commit": {
                "author": {
                    "name": "Vishwas Shashidhar",
                    "email": "[hidden]",
                    "date": "2017-09-28T13:12:47Z"
                },
                "committer": {
                    "name": "Vishwas Shashidhar",
                    "email": "[hidden]",
                    "date": "2017-09-28T13:12:47Z"
                },
                "message": "electron-145: fixes the issue with invalid json config upon repair",
                "tree": {
                    "sha": "a6019abb4fdac84dfcdc56572b128429e8d91b37",
                    "url": "https://api.github.com/repos/symphonyoss/SymphonyElectron/git/trees/a6019abb4fdac84dfcdc56572b128429e8d91b37"
                },
                "url": "https://api.github.com/repos/symphonyoss/SymphonyElectron/git/commits/1ff558af8121487cdbc0302b4b8efb8e930b33af",
                "comment_count": 0
            },
            "url": "https://api.github.com/repos/symphonyoss/SymphonyElectron/commits/1ff558af8121487cdbc0302b4b8efb8e930b33af",
            "html_url": "https://github.com/symphonyoss/SymphonyElectron/commit/1ff558af8121487cdbc0302b4b8efb8e930b33af",
            "comments_url": "https://api.github.com/repos/symphonyoss/SymphonyElectron/commits/1ff558af8121487cdbc0302b4b8efb8e930b33af/comments",
            "author": null,
            "committer": null,
            "parents": [
                {
                    "sha": "1a312544d7223b7aa041219cd91958f83bfa235c",
                    "url": "https://api.github.com/repos/symphonyoss/SymphonyElectron/commits/1a312544d7223b7aa041219cd91958f83bfa235c",
                    "html_url": "https://github.com/symphonyoss/SymphonyElectron/commit/1a312544d7223b7aa041219cd91958f83bfa235c"
                }
            ]
        }
    ]
}
```

**Comment by maoo** [comment]
It seems that the issue is caused by the Github user email not being associated with the Github profile - https://github.com/octokit/octokit.net/issues/1433#issuecomment-235714712 .

To fix it, the commit author must run `git config --list | grep email` locally and make sure that the email address is listed in https://github.com/settings/emails , then add an issue comment to re-summon the CLA bot check (`@finos-cla-bot[bot] check` in our case).

We may want to build a fallback strategy by following the advise on https://github.com/octokit/octokit.net/issues/1433#issuecomment-235710790

**Comment by ColinEberhardt** [comment]
Good spot - I'll put in place a fallback shortly. Thanks for including the log information, very useful!

**Comment by ColinEberhardt** [comment]
fixed by #106
